### PR TITLE
fix(GetStatusBackend)_: Made sure BackupDisabledDataDir is always a valid absolute path.

### LINKED ISSUE #5186: Cannot login on Windows
# Problem

During login of existing account, or creation o new account or syncing between two apps, Windows desktop client will error out and not login 

image

logs:

```
DEBUG[05-17|21:05:54.175|github.com/status-im/status-go/services/wallet/collectibles/commands.go:340]       end loadOwnedCollectiblesCommand         chain=10 account=0x4d910637189E01cbd5f3BF310f92Ed7993DC707f in=593.4971ms
DEBUG[05-17|21:05:54.185|github.com/status-im/status-go/node/get_status_node.go:510]                        status node stopped                      package=status-go/node.StatusNode
ERROR[05-17|21:05:54.185|github.com/status-im/status-go/mobile/status.go:319]                               loginAccount failed                      error="mkdir D:\\Projects\\status-desktop\\Status\\data\\D:: The filename, directory name, or volume label syntax is incorrect."
``` 

# Notes
It looks like 
vendor\status-go\api\geth_backend.go
```
conf.ShhextConfig.BackupDisabledDataDir = filepath.Join(b.rootDataDir, conf.ShhextConfig.BackupDisabledDataDir)
```
is causing this problem, because in 
vendor\status-go\services\ext\service.go
```
func (s *Service) InitProtocol(nodeName string, identity *ecdsa.PrivateKey, appDb, walletDb *sql.DB, httpServer *server.MediaServer, multiAccountDb *multiaccounts.Database, acc *multiaccounts.Account, accountManager *account.GethManager, rpcClient *rpc.Client, walletService *wallet.Service, communityTokensService *communitytokens.Service, wakuService *wakuv2.Waku, logger *zap.Logger) error {
....
	dataDir := filepath.Clean(s.config.ShhextConfig.BackupDisabledDataDir)

	if err := os.MkdirAll(dataDir, os.ModePerm); err != nil {
		return err
	}

...
```

there dataDir  will contain 'self-duplicated' path (in my case it is  D:\Projects\status-desktop\Status\data\ D:\Projects\status-desktop\Status\data\

Commenting out ```conf.ShhextConfig.BackupDisabledDataDir = filepath.Join(b.rootDataDir, conf.ShhextConfig.BackupDisabledDataDir)
``` locally will make the app to properly login and ```dataDir := filepath.Clean(s.config.ShhextConfig.BackupDisabledDataDir)``` contain proper path

### PR DESCRIPTION
**Pull request by Seitseman**
In GethStatusBackend loadNodeConfig sometimes BackupDisabledDataDir is passed as absolute path and sometimes as relative "./" . And `filepath.Join` will always concatenate its arguments which may result on Windows in invalid pathname. Later in the code, Mkdir will fail and prevent new users from logging in, not allow device sync from other app instances, and may not allow current user to login.
I've tried to modify all BackupDisabledDataDir  to be either absolute paths or relative paths, but those changes did not cover all needed use cases.

Checking PR https://github.com/status-im/status-desktop/pull/14930 that fixes some of the config issue proved that it does not fix BackupDisabledDataDir  problem.
Although this fix is might be far from optimal solution, it at least allows new users to be able to login on Windows' Status client

Closes #5186

### CODE REVIEW

File: api/geth_backend.go
Comment (igor-sirotin) [inline api/geth_backend.go]: Do we really need this fallback? Is it safe? 🤔

**Review by igor-sirotin — COMMENTED** [review]
Thanks for the fix!

I think we should cover this we a test. `TestSetFleet` or `TestWalletConfigOnLoginAccount` can be used as a base.
Both of them call `LoginAccount`, which eventually calls `loadNodeConfig`.

**Review by jrainville — COMMENTED** [review]
Good job on finding a fix. I'll wait for your answer to Igor before approving, but otherwise, it looks good.

Also, once it's merged, please create a new PR that cherry-picks this commit to the `release/0.179.x` branch.

File: api/geth_backend.go
Comment (saledjenic) [inline api/geth_backend.go]: Do we need to return an error here instead of letting the flow continue?

File: api/geth_backend.go
Comment (saledjenic) [inline api/geth_backend.go:1755]: How does this approach differ from handling `KeyStoreDir`? Do we need to do the same check for `KeyStoreDir`?

File: api/geth_backend.go
Comment (saledjenic) [inline api/geth_backend.go]:
```suggestion
backupDir := conf.ShhextConfig.BackupDisabledDataDir
if !filepath.IsAbs(backupDir) {
    backupDir = filepath.Join(b.rootDataDir, backupDir)
    
    if !filepath.IsAbs(backupDir) {
        b.log.Error("BackupDisabledDataDir was expected to be an absolute path, instead it is ", backupDir)
        return errors.New("BackupDisabledDataDir was expected to be an absolute path")
    } 
    conf.ShhextConfig.BackupDisabledDataDir = backupDir
}
```

**Review by jrainville — APPROVED** [review]
It would be good to have a PR on status-desktop so that it can be tested on all OSes to make sure everything is still working

**Comment by jrainville** [comment]
FYI, I rebased this PR and also created the desktop equivalnent here for QAs to test: https://github.com/status-im/status-desktop/pull/15007

Mobile one here: https://github.com/status-im/status-mobile/pull/20296

**Comment by jrainville** [comment]
@Seitseman can you look at this PR again. It seems both desktop and mobile have issues with those changes. see comment here for the PRs https://github.com/status-im/status-go/pull/5255#issuecomment-2142677117

**Comment by igor-sirotin** [comment]
I'm taking a quick look at this as well

**Comment by igor-sirotin** [comment]
It seems to me that we should specify the for of `BackupDisabledDir` (absolute/relative) and give an error when no match, instead of trying to make the `loadNodeConfig` smart.

Also it seems it was me who broke this here: https://github.com/status-im/status-desktop/pull/14139

**Comment by igor-sirotin** [comment]
Closing in favour of https://github.com/status-im/status-go/pull/5284

**Comment by igor-sirotin** [comment]
Closing in favour of https://github.com/status-im/status-go/pull/5284

**Pull request closed without merging**
